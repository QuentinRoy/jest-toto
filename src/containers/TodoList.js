import React, { PropTypes } from 'react'
import { connect } from 'react-redux'
import Todo from '../components/Todo'
import { toggleTodo } from '../actions'

const TodoList = ({ todos, onTodoClick }) => (
  <ul>
    {todos.map(todo =>
      <Todo
        key={todo.id}
        {...todo}
        onClick={() => onTodoClick(todo.id)}
      />
    )}
  </ul>
)

TodoList.propTypes = {
  todos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    completed: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired).isRequired,
  onTodoClick: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({ todos: state })

const mapDispatchToProps = {
  onTodoClick: toggleTodo
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoList)
