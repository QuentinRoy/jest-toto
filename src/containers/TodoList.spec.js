import React from 'react'
import renderer from 'react-test-renderer'
import TodoList from './TodoList'
import configureMockStore from 'redux-mock-store'

// Mock the Todo module and replace its export by a <Todo> tag.
jest.mock('../components/Todo', () => 'Todo')
// Also mock the toggleTodo action from the actions module.
jest.mock(
  '../actions',
  () => ({ toggleTodo: jest.fn()
}));

const mockStore = configureMockStore()

describe('<TodoList />', () => {
  it('renders correctly', () => {
    // Define the initial state.
    const state = [
      { id: 1, text: 'foo', completed: false },
      { id: 2, text: 'bar', completed: true }
    ]
    // Create the store from this state.
    const store = mockStore(state)
    // Renders <TodoList> with this store.
    const tree = renderer.create(
      <TodoList
        store={store}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
