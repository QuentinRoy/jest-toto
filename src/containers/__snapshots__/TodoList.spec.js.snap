exports[`<TodoList /> renders correctly 1`] = `
<ul>
  <Todo
    completed={false}
    id={1}
    onClick={[Function]}
    text="foo" />
  <Todo
    completed={true}
    id={2}
    onClick={[Function]}
    text="bar" />
</ul>
`;
